### PR TITLE
Course browser by subject

### LINKED ISSUE #519: Navigation Fixes for Curriculum Pages
Added proper back buttons to the Course Grouping related pages.

[Story Points: 1]
[Priority: Low]
[Risk: Low]

### LINKED ISSUE #520: Allow Reviewers to Edit the Name and Description of a Dossier
As a reviewer for a dossier, I want to be able to edit the dossier's name and description during the review process.

[Story Points: 1]
[Priority: Low]
[Risk: Medium]

Demo steps,
View dossiers for review,
Click on the edit button,
Enter new title or description,
Submit changes.

### LINKED ISSUE #521: View all Courses with Shared Subject Browser
As a user I want to see a list of all courses of a given subject and be able to access their details page from there.

[Story Points: 4]
[Priority: Low]
[Risk: Low]

Demo Steps:
Go to the Browser List page and Click on the Course by Subject Browser,
Enter the subject you're looking for,
Click enter,
Navigate through the list to find the course you're looking for,
Click on the details button to view the course's details.

### PR DESCRIPTION
**Pull request by Dinferno-lol**
Adds a course by subject browser which allows users to view a list of all courses of a given subject. Also includes navigation fixes and allows reviewers to edit the name and description of a dossier.

closes #521 
closes #520 
closes #519

### CODE REVIEW

File: App/src/pages/dossier/Dossiers.tsx [hunk 1]
```diff
@@ -257,6 +257,7 @@ export default function Dossiers() {
                             handleNavigateToDossierReview={handleNavigateToDossierReview}
                             useIcons={false}
                             reviewIcons={true}
+                            editable={true}
```
Comment (JamesPartsafas) [inline App/src/pages/dossier/Dossiers.tsx:260]: This is not where we want it. You should not be able to edit your dossiers once they've been submitted. Instead, the page that displays Dossiers Requiring Your Review should be the one with this edit button.

**Comment by JamesPartsafas** [comment]
Because you added a new button for the subject browser, we have a redundant button on the Browser page that doesn't actually work when clicked that should be removed. 
image

**Comment by JamesPartsafas** [comment]
Everything looks good except the 2 bits of feedback I gave, nice work.

File: App/src/pages/dossier/Dossiers.tsx (hunk 1 above)
Comment (Dinferno-lol) [reply App/src/pages/dossier/Dossiers.tsx:260]: Ok, sorry for the error, I fixed it now.

Comment (Dinferno-lol) [reply App/src/pages/dossier/Dossiers.tsx:260]: I also got rid of the unneeded button with my last commit

**Review by JamesPartsafas — APPROVED** [review]

**Pull request merged**